projects/observability/src/shared/components/table/data-cell/exit-calls/exit-calls-table-cell-renderer.component.ts

```diff
@@ -0,0 +1,74 @@
+import { ChangeDetectionStrategy, Component, Inject, OnInit } from '@angular/core';
+import { Dictionary } from '@hypertrace/common';
+import {
+  CoreTableCellParserType,
+  TableCellAlignmentType,
+  TableCellParserBase,
+  TableCellRenderer,
+  TableCellRendererBase,
+  TableColumnConfig,
+  TABLE_CELL_DATA,
+  TABLE_COLUMN_CONFIG,
+  TABLE_COLUMN_INDEX,
+  TABLE_DATA_PARSER,
+  TABLE_ROW_DATA
+} from '@hypertrace/components';
+import { Trace } from '@hypertrace/distributed-tracing';
+import { ObservabilityTableCellType } from '../../observability-table-cell-type';
+
+interface CellData {
+  units: number;
+  value: [number, Dictionary<string>];
+}
+@Component({
+  selector: 'ht-exit-calls-table-cell-renderer',
+  styleUrls: ['./exit-calls-table-cell-renderer.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="exit-calls-cell" [htTooltip]="exitCallsTooltip">
+      <span class="exit-calls-count">{{ this.apiExitCalls }}</span>
+
+      <ng-template #exitCallsTooltip>
+        <ng-container *ngIf="this.apiExitCalls > 0">
+          <div *ngFor="let item of this.apiCalleeNameCount" class="api-callee-name-count">
+            <span class="api-callee-name">{{ item[0] }}</span>
+            <span class="api-callee-count">{{ item[1] }}</span>
+          </div>
+          <div
+            *ngIf="this.totalCountOfDifferentApiCallee > this.maxShowApiCalleeNameCount"
+            class="remaining-api-callee"
+          >
+            and {{ this.totalCountOfDifferentApiCallee - this.maxShowApiCalleeNameCount }} more
+          </div>
+        </ng-container>
+        <ng-container *ngIf="this.apiExitCalls <= 0" class="no-exit-calls">No exit calls</ng-container>
+      </ng-template>
+    </div>
+  `
+})
+@TableCellRenderer({
+  type: ObservabilityTableCellType.ExitCalls,
+  alignment: TableCellAlignmentType.Left,
+  parser: CoreTableCellParserType.NoOp
+})
+export class ExitCallsTableCellRendererComponent extends TableCellRendererBase<CellData, Trace> implements OnInit {
+  public readonly apiCalleeNameCount: string[][];
+  public readonly apiExitCalls: number;
+  public readonly maxShowApiCalleeNameCount: number = 10;
+  public readonly totalCountOfDifferentApiCallee!: number;
+
+  public constructor(
+    @Inject(TABLE_COLUMN_CONFIG) columnConfig: TableColumnConfig,
+    @Inject(TABLE_COLUMN_INDEX) index: number,
+    @Inject(TABLE_DATA_PARSER)
+    parser: TableCellParserBase<CellData, Trace, unknown>,
+    @Inject(TABLE_CELL_DATA) cellData: CellData,
+    @Inject(TABLE_ROW_DATA) rowData: Trace
+  ) {
+    super(columnConfig, index, parser, cellData, rowData);
+    const apiCalleeNameCount: string[][] = Object.entries(cellData.value[1]);
```

Comment:
- This name is confusing to me mainly because of 'Count' suffix. So  `apiCalleeNameCount` is a dictionary? 

can we call it `apiCalleeNameEntires` instead? 

- Is `[string, string][]` a better type? 

- totalCountOfDifferentApiCallee -> uniqueApiCallee

- Line 71. Why do we need to use `slice` here?

Reply:
Agree naming is confusing/type suggestions. Line 71 is meant to limit the amount of pairs to show in the tooltip, but again naming would help clarify (`maxShowApiCalleeNameCount` as a const value could be `MAX_API_CALLEE_TO_SHOW`, for example).

Reply:
Let's also rename `maxShowApiCalleeNameCount` to `MAX_API_CALLEE_TO_SHOW`